packages/expo-screen-orientation/app.plugin.js

```diff
@@ -0,0 +1,62 @@
+const pkg = require('./package.json');
+const { createRunOncePlugin, withDangerousMod, IOSConfig } = require('@expo/config-plugins');
+const fs = require('fs-extra');
+const assert = require('assert');
+
+const OrientationLock = {
+  DEFAULT: 'UIInterfaceOrientationMaskAllButUpsideDown',
+  ALL: 'UIInterfaceOrientationMaskAll',
+  PORTRAIT: 'UIInterfaceOrientationMaskPortrait',
+  PORTRAIT_UP: 'UIInterfaceOrientationMaskPortrait',
+  PORTRAIT_DOWN: 'UIInterfaceOrientationMaskPortraitUpsideDown',
+  LANDSCAPE: 'UIInterfaceOrientationMaskLandscape',
+  LANDSCAPE_LEFT: 'UIInterfaceOrientationMaskLandscapeLeft',
+  LANDSCAPE_RIGHT: 'UIInterfaceOrientationMaskLandscapeRight',
+};
+
+const withScreenOrientationViewController = (config, { initialOrientation = 'DEFAULT' } = {}) => {
+  assert(
+    initialOrientation in OrientationLock,
+    `Invalid initial orientation "${initialOrientation}" expected one of: ${Object.keys(
+      OrientationLock
+    ).join(', ')}`
+  );
+
+  return withDangerousMod(config, [
+    'ios',
+    async config => {
+      const fileInfo = IOSConfig.Paths.getAppDelegate(config.modRequest.projectRoot);
+      let contents = await fs.readFile(fileInfo.path, 'utf-8');
+      if (fileInfo.language === 'objc') {
+        // Add import
+        if (
+          !contents.includes('#import <EXScreenOrientation/EXScreenOrientationViewController.h>')
+        ) {
+          contents = contents.replace(
+            /\#import \"AppDelegate.h\"/g,
+            `#import "AppDelegate.h"
+#import <EXScreenOrientation/EXScreenOrientationViewController.h>`
+          );
+        }
+
+        // Change View Controller
+        if (!contents.includes('[EXScreenOrientationViewController alloc]')) {
+          contents = contents.replace(
+            /UIViewController\s?\*\s?rootViewController\s?=\s?\[UIViewController new\]\;/g,
```

Comment:
this will work fine for debug mode but it looks like we have some incompatibility between expo-updates and expo-screen-orientation, they both want to control the root view controller. to complicate things again, so does expo-dev-launcher, and it wouldn't surprise me if we had more cases. @EvanBacon raised this as something we need to solve for, but i won't block this pr on it

Reply:
ya we might need something like a Unimodule ViewController which can proxy its methods to unimodules.